stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/preview/internal/StreamVideoPlayerContent.kt

```diff
@@ -0,0 +1,255 @@
+ */
+
+package io.getstream.chat.android.compose.ui.attachments.preview.internal
+
+import android.content.Context
+import androidx.annotation.OptIn
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.compose.LifecycleResumeEffect
+import androidx.media3.common.MediaItem
+import androidx.media3.common.PlaybackException
+import androidx.media3.common.Player
+import androidx.media3.common.util.UnstableApi
+import androidx.media3.exoplayer.ExoPlayer
+import androidx.media3.ui.PlayerView
+import io.getstream.chat.android.compose.R
+import io.getstream.chat.android.compose.ui.attachments.content.PlayButton
+import io.getstream.chat.android.compose.ui.components.LoadingIndicator
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.compose.ui.util.StreamAsyncImage
+import io.getstream.chat.android.compose.ui.util.clickable
+
+/**
+ * A Composable that displays a video player with support for thumbnails, play controls, and buffering indicators.
+ *
+ * This component creates a Media3 ExoPlayer instance that handles video playback while respecting
+ * the Android lifecycle. The player is automatically created when the composable enters the composition
+ * and released when it leaves or pauses.
+ *
+ * The component offers several features:
+ * - Shows an optional thumbnail with play button before playback starts
+ * - Displays media controls that can be shown initially or hidden
+ * - Shows a buffering indicator during loading states
+ * - Automatically handles player lifecycle and resource cleanup
+ *
+ * @param assetUrl The URL of the video to play.
+ * @param playWhenReady Whether playback should start automatically after preparation.
+ * @param showControllerInitially Whether to show the player controls when first displayed.
+ * @param onPlaybackError Callback invoked when video playback encounters an error.
+ * @param modifier Modifier to be applied to the player container.
+ * @param thumbnailUrl Optional URL of the thumbnail image to display before playback starts.
+ * @param thumbnailEnabled Whether to show the thumbnail before playback starts.
+ */
+@OptIn(UnstableApi::class)
+@Composable
+internal fun StreamVideoPlayerContent(
```

Comment:
This component isn't also used to play audio?
Maybe its name could be `StreamMediaPlayerContent`, WDYT?